modules/tensor_mechanics/include/materials/Compute2DIncrementalStrain.h

```diff
@@ -0,0 +1,30 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef COMPUTE2DINCREMENTALSTRAIN_H
+#define COMPUTE2DINCREMENTALSTRAIN_H
+
+#include "ComputeIncrementalSmallStrain.h"
+
+/**
+ * Compute2DIncrementalStrain defines a strain increment only for
+ * incremental strains in 2D geometries, handling the out of plane strains.
+ * Compute2DIncrementalStrain contains a virtual method to define the strain_zz
+ * as a nonzero value in the inherited classes ComputePlaneIncrementalStrain
+ * ComputeGeneralizedPlaneIncrementalStrain and ComputeAxisymmetricRZIncrementalStrain.
+ */
+class Compute2DIncrementalStrain : public ComputeIncrementalSmallStrain
+{
+public:
+  Compute2DIncrementalStrain(const InputParameters & parameters);
+
+protected:
+  virtual void computeTotalStrainIncrement(RankTwoTensor & total_strain_increment, unsigned int qp);
+  virtual Real computeDeformGradZZ(unsigned int qp) = 0;
+  virtual Real computeDeformGradZZold(unsigned int qp) = 0;
```

Comment:
It looks like _none_ of the above methods needs the `qp` argument. Just use `_qp` in those methods!